beacon_node/lighthouse_network/src/discovery/mod.rs

```diff
@@ -925,31 +926,94 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
 impl<TSpec: EthSpec> NetworkBehaviour for Discovery<TSpec> {
     // Discovery is not a real NetworkBehaviour...
```

Comment:
FWIW, we see `NetworkBehaviour`s as plugins and `mDNS` for example is also a `NetworkBehaviour` so implementing discovery as one is totally fine.